### ISSUE TITLE
How to experiment different learning rates with HParams Dashboard?

### ISSUE THREAD

**Issue by arwen-x** [tensorflow/tensorboard#3688]
I want to see how different learning rates affects my model's training.

```
HP_NUM_UNITS = hp.HParam('num_units', hp.Discrete([16, 32]))
HP_OPTIMIZER = hp.HParam('optimizer', hp.Discrete(['adam', 'sgd']))
HP_DROPOUT = hp.HParam('dropout', hp.RealInterval(0.1, 0.2))
HP_L_RATE= hp.HParam('learning_rate', hp.Discrete([0.0005, 0.001]))
```

I can not figure out where and how to insert hparams[HP_L_RATE] in the tensorflow model, for example the model given in the [documentation](https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams):

```
def train_test_model(hparams):
  model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(hparams[HP_NUM_UNITS], activation=tf.nn.relu),
    tf.keras.layers.Dropout(hparams[HP_DROPOUT]),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax),
  ])
  model.compile(
      optimizer=hparams[HP_OPTIMIZER],
      loss='sparse_categorical_crossentropy',
      `metrics=['accuracy'],`
  )

  model.fit(x_train, y_train, epochs=1) # Run with 1 epoch to speed things up for demo purposes
  _, accuracy = model.evaluate(x_test, y_test)
  return accuracy
```
Thanks,

**Comment by wchargin** [comment]
Hi @arwen-x! Thanks for reaching out.

In general, you want to plumb the value of the hparam into the part of
the model that actually controls that hyperparameter. So this question
boils down to, “in Keras, how do I control the learning rate?”

In Keras, the learning rate is a parameter of the optimizer. When you
call `model.compile`, the `optimizer` keyword argument can be either a
string name like `"adam"` or an optimizer like an `AdamOptimizer`. (The
meaning of `"adam"` is just “create an `AdamOptimizer` with the default
parameters”.) So in this case you should create an optimizer based on
the desired `HP_OPTIMIZER` and `HP_L_RATE` hparams:

```python
optimizer_name = hparams[HP_OPTIMIZER]
learning_rate = hparams[HP_L_RATE]
if optimizer_name == "adam":
    optimizer = tf.keras.optimizers.Adam(learning_rate=learning_rate)
elif optimizer_name == "sgd":
    optimizer = tf.keras.optimizers.SGD(learning_rate=learning_rate)
else:
    raise ValueError("unexpected optimizer name: %r" % (optimizer_name,))
model.compile(
    optimizer=optimizer,
    loss='sparse_categorical_crossentropy',
    `metrics=['accuracy'],`
)
```

Does this make sense?

**Comment by arwen-x** [comment]
Thank you!

**Comment by wchargin** [comment]
You’re quite welcome.
